tests/test_experiments.py

```diff
@@ -228,12 +229,19 @@ def sample_config():
         'temporal_config': temporal_config,
         'grid_config': grid_config,
         'scoring': scoring_config,
-        'user_metadata': { 'custom_key': 'custom_value' },
+        'user_metadata': {'custom_key': 'custom_value'},
         'individual_importance': {'n_ranks': 2}
     }
 
 
-def simple_experiment_test(experiment_class):
+parametrize_experiment_classes = pytest.mark.parametrize(('experiment_class',), [
+    (SingleThreadedExperiment,),
+    (partial(MultiCoreExperiment, n_processes=2, n_db_processes=2),),
+])
+
+
+@parametrize_experiment_classes
+def test_simple_experiment(experiment_class):
```

Comment:
i'm curious, how does this work from the py.test command-line? Under the current regime, I'm used to being able to run one test like this:
`py.test tests/test_experiments.py::test_singlethreaded_experiment`

Clearly this isn't the usual invocation, but I find it useful for debugging. Is there a way to just run one of the tests from py.test CLI?

Reply:
This was annoying to track down, but yes, you may continue to select individual test cases in this way, by referring to the values they'll take. `pytest` either accepts the string representation of the value, when appropriate, or computes a value name, like `experiment_class1`.

You can see how to refer to the cases by simply running with `--collect-only`:

    $ py.test tests/test_experiments.py -k test_simple_experiment --collect-only
    ================================= test session starts ==================================
    platform linux -- Python 3.5.2, pytest-2.9.2, py-1.4.34, pluggy-0.3.1
    rootdir: /home/jester/dev/repo/dsapp/triage, inifile: pytest.ini
    plugins: cov-2.5.1
    collected 14 items
    <Module 'tests/test_experiments.py'>
      <Function 'test_simple_experiment[SingleThreadedExperiment]'>
      <Function 'test_simple_experiment[experiment_class1]'>

    ================== 12 tests deselected by '-ktest_simple_experiment' ===================
    ============================ 12 deselected in 0.90 seconds =============================

And so, this works:

    $ py.test tests/test_experiments.py -v -k "test_simple_experiment[experiment_class1]"    ================================= test session starts ==================================        platform linux -- Python 3.5.2, pytest-2.9.2, py-1.4.34, pluggy-0.3.1 -- /home/jester/.virtualenvs/triage/bin/python3    cachedir: .cache
    rootdir: /home/jester/dev/repo/dsapp/triage, inifile: pytest.ini
    plugins: cov-2.5.1
    collected 14 items

    tests/test_experiments.py::test_simple_experiment[experiment_class1] PASSED

    ========= 13 tests deselected by '-ktest_simple_experiment[experiment_class1]' =========        ======================= 1 passed, 13 deselected in 9.47 seconds ======================== 

As well as with the shorter syntax:

    $ py.test -v tests/test_experiments.py::test_simple_experiment[experiment_class1]
    ================================= test session starts ==================================    platform linux -- Python 3.5.2, pytest-2.9.2, py-1.4.34, pluggy-0.3.1 -- /home/jester/.virtualenvs/triage/bin/python3    cachedir: .cache
    rootdir: /home/jester/dev/repo/dsapp/triage, inifile: pytest.ini
    plugins: cov-2.5.1
    collected 13 items

    tests/test_experiments.py::test_simple_experiment[experiment_class1] PASSED

    =============================== 1 passed in 9.21 seconds ==============================